packages/contracts/test/token/erc20/governance-wrapped-erc20.ts

```diff
@@ -32,503 +34,563 @@ let to: SignerWithAddress;
 let other: SignerWithAddress;
 let toDelegate: string;
 
+
 describe('GovernanceWrappedERC20', function () {
-  let signers: SignerWithAddress[];
-  let governanceToken: GovernanceWrappedERC20;
-  let erc20: TestERC20;
-  let TestERC20: TestERC20__factory;
-  let GovernanceWrappedERC20: GovernanceWrappedERC20__factory;
-  let defaultBalances: AccountBalance[];
-
-  let defaultExistingERC20InitData: [string, string, number];
-  let defaultGovernanceWrappedERC20InitData: [string, string, string];
-
-  before(async () => {
-    signers = await ethers.getSigners();
-
-    TestERC20 = new TestERC20__factory(signers[0]);
-    GovernanceWrappedERC20 = new GovernanceWrappedERC20__factory(signers[0]);
-    defaultBalances = [
-      {account: signers[0].address, amount: 123},
-      {account: signers[1].address, amount: 456},
-      {account: signers[2].address, amount: 789},
-    ];
-
-    from = signers[0];
-    to = signers[1];
-    other = signers[2];
-  });
+    let signers: SignerWithAddress[];
+    let governanceToken: GovernanceWrappedERC20;
+    let erc20: TestERC20;
+    let dao: DAO;
+    let TestERC20: TestERC20__factory;
+    let GovernanceWrappedERC20: GovernanceWrappedERC20__factory;
+    let defaultBalances: AccountBalance[];
+
+    let defaultExistingERC20InitData: [string, string, number];
+    let defaultGovernanceWrappedERC20InitData: any[] = [];
+
+
+    before(async () => {
+      signers = await ethers.getSigners();
+
+      TestERC20 = new TestERC20__factory(signers[0]);
+      GovernanceWrappedERC20 = new GovernanceWrappedERC20__factory(signers[0]);
+      defaultBalances = [
+        {account: signers[0].address, amount: 123},
+        {account: signers[1].address, amount: 456},
+        {account: signers[2].address, amount: 789},
+      ];
+
+      from = signers[0];
+      to = signers[1];
+      other = signers[2];
+    });
 
-  beforeEach(async function () {
-    defaultExistingERC20InitData = [existingErc20Name, existingErc20Symbol, 0];
-    erc20 = await TestERC20.deploy(...defaultExistingERC20InitData);
+    beforeEach(async function () {
+      defaultExistingERC20InitData = [
+        existingErc20Name,
+        existingErc20Symbol,
+        0,
+      ];
+      erc20 = await hre.wrapper.deploy('TestERC20', {
+        args: defaultExistingERC20InitData,
+      });
 
-    const promises = defaultBalances.map(balance =>
-      erc20.setBalance(balance.account, balance.amount)
-    );
-    await Promise.all(promises);
+      dao = await deployNewDAO(signers[0]);
 
-    defaultGovernanceWrappedERC20InitData = [
-      erc20.address,
-      governanceWrappedERC20Name,
-      governanceWrappedERC20Symbol,
-    ];
+      for (let i = 0; i < defaultBalances.length; i++) {
+        await erc20.setBalance(
+          defaultBalances[i].account,
+          defaultBalances[i].amount
+        );
+      }
```

Comment:
`Promise.all` is much quicker than `await`ing every `setBalance`. This `beforeEach` setup will be repeated a lot since it's heavily nested.

Reply:
Really? What the heck?